client/components/app/component.js

```diff
@@ -0,0 +1,8 @@
+//mainComponent
+var mainComponent = FlowComponents.define('main',function(props){
+  this.set('template',props.template)
```

Comment:
You can directly set this as `componentName`. Then no need to define a separate state.
Also, try to change the prop name also into `componentName`